.travis.yml

```diff
@@ -0,0 +1,232 @@
+  script:
+    - echo 'Running build stage on docker container $DOCKER_CONTAINER_ID'
+    - docker exec -ti $DOCKER_CONTAINER_ID /bin/bash -ec "cd /home/daqbuild/gem-plotting-tools; make && make rpm"
+    - docker exec -ti $DOCKER_CONTAINER_ID /bin/bash -ec "cd /home/daqbuild/gem-plotting-tools; rpm -q --filesbypkg -p rpm/*.rpm"
+    - ls -laZ rpm
+    - find ./rpm -iname '*.rpm' -o -iname '*.tar.gz' -o -iname '*.tbz2'
+  after_success:
+    - echo "Testing the installation of the package"
+    - virtualenv building -p python --system-site-packages
+    - . building/bin/activate
+    - pip install rpm/*.zip
+    - echo "Install the dependencies"
+    - echo "Run the predefined test"
+    - ./.ci/docker.sh stop ${DOCKER_IMAGE}
+
+.release: &release_template
+  skip_cleanup: true
+  # publish to Github releases
+  provider: releases
+  api_key:
+    # secure: "l3w7DGKblwk+XRj9jDKLNfz99BXhyScEN2T6toDca3d1YQaCmbCaJzFYjxkdCuu70/uZtI2NbWfAbL6vjqMaqJqa75xxHguz1gDqOYro5SJQ1nV8MWTPH39qiUBRZEXE61CiDJTAree+l1Br0hvIaaNXmW2V+m8hpOzHy+V15HS2aanKUr0XSwZgzuCkKxapY5Ha2ippdniiW6GVt9Tr4UK1qFlBZPSK/vHxPrs/7bJzVPhsSKD66WJo1tHlBZTCjmEBMdsN3GlMSmcq6tEv37Ugu7/NMB79JQ3Ccyc5ty7yw1m+gi1EDQRuA0Ztf+IAJU8KYtCjGvFLSirDvIBFGmwQhY3CyFDg20RBlX32p+b7JvzsLFMc/9nkOmLlTpWr68Eb/HZSOabEiR/ETufloQx9RUwa+LFYsmfNMywdzedckenYHRltJP+8zKuSAiQeaHJWuCFCP0irRhTCgFLSZy+GFeWiP8tH/Xj1ZbCBP9Rulr3Jy99rqme7iflcxAqnjgmFAOKbwlXUV47P/QKh/Ec2jb9dNrJPQ3WWkP9va6MdQo9C+cfg/6du0UmMlNmALX9DMCWTRJcR1OtYKVrgeDJ67V6YJpqi9kXdWhvYORUBdJkkazJwparRm7FPnKXg0fThdlSAKUaN4P+6uDwZLYL5ispINUmkyBF1TZ89OyU="
+    # secure: "LKzmbCOo8PGN+MXe1EE1AjxxxejQa9GuJ3e3J67B2o3JWTAEZmH6tawPcChj1bSFyCpLbEnOLAOhHSRID6lB136pos2/TXpQZncMWWgO+SYLMZwiK4GDCAGPZ9cxgkOQZGm7OacXxTbpEVen8OTNU5M37+zN06dHUgfoU7lZk6M66h56BIuxNgc+ViucsUmzc3cZXlynglWvR0LX+Vlf2Gh8OsyJXDFDfKxEcWqv5QW1oyTVUowvseO7QF5jdgI12K9EGygplbXkVjsPsJem37HnaZTnukdG1CTt2v7aP56c6UdgsXDbhB1XLzcIh0x50liGLpEhWWmhOU2SMmy4XxhXLpjCT5Q92SahLjX28Y1A753nSZGDCSSNYOYpu2X8rQ9/jxHhQsEj5uHTB0Ht2DLJ8rq3DhacRinokXoCqulmIhJ4oztXRn3KIXOvnrl95m4dmHuYoVndV/2TmDtlOt/6LHvtXHoePi0yDS+ukI+c21uJ0iPxE+4dLToo05K8SdHbGu0x1NaGyN/IJV2CgZzrD0sBiZggNh5u7UcdsJpG9qVuNmpJenMWT/LfMynhkSCyDMs9xz0MwQkiNnpsfzGtn6P2vZBxlJkAFI/UXvvzZf1nR7x12JTAW7z8BAUSxefXM3Uw0zbQtriNkHLmWERzK6tBRUi6d1FZU34Z02I="
+    secure: "IZG4HXpAMFkAWzEnOG52/D3qg405xiNZQtVd+B6swV9Bb6/2/KGhPYWyb6ky06x83ZXZUtKG5QqE5/r7ohjcKuP1b25A3sSKr1iVdv9+8pRjahQ3aXWzQftmWW49ZAa/2GSwn+pHgXSHNcNW+CfidOSwCC6owt4RQxZuppoSjF7PeVWMUhNrwGukws0/jpBytahuhLIO4lOb4ShlsQqNjzkbmEsE2z0+zWpDeHN/sz0U71jFzaIG90Ygu/5huemS4g+jo7MWqtKVlKXe3j1qPgN15Wg5Rnm/osRwPpPTQz6MTOehn4WswYwYwGoMQ9I97RhxbB3QszRJHl4L7OxNJIcY2Pr3Nk9iRuvBSsdmgFBA8l2pxD14xCp1st4UfU35aQm84yG2VuLwys8KFjNFNap71BdoVQoZISThlagMKG6oWVLri1vfRaIcEJjT+s6blcK0CZ9xhMd6lVnYTOwDdJuIv3KcsQkzvF/9iiENkuH5PUU9HLVUw5/Nxjq3JoNzs9+ohIpgNLT4vNOMq5ZewCib5GDnIq732gOcuqNRZMTCRKY+/IgcPJJv935yGlcGiJ246Al1jhSS45rcvazgFty4NRwM41McaN83ZPqvqRLDRI4hRjlaEmb6fnDIZJbOLldJYjOOJxm+FbhPzWiYOOpjKllj/xQ7x0lsjLXEubY="
```

Comment:
again, should the api key be available in public?

Reply:
See https://docs.travis-ci.com/user/encryption-keys
* provided no one ever (maliciously or otherwise) attempts to expose these, it's secure (up to the level that Travis can ensure it)
  * If I ever suspect a hole, I just deauthorize the given key

Reply:
That was a good guide on how to do this, but what exactly are we trying to encrypt and keep secret...?  This is relevant question since a developer should know when and what to encrypt moving forward.

Reply:
The encryption is **only** relevant for making a set of released binaries, **only** the maintainer(s) will be responsible for this